gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -278,18 +320,29 @@ public void copy(Path source, Path target, CopyOption... options) throws IOExcep
       } else if (option instanceof CloudStorageOption) {
         if (option instanceof OptionBlockSize) {
           blockSize = ((OptionBlockSize) option).size();
+        } else if (option instanceof OptionMimeType) {
+          tgtInfoBuilder.contentType(((OptionMimeType) option).mimeType());
+          setContentType = true;
+        } else if (option instanceof OptionCacheControl) {
+          tgtInfoBuilder.cacheControl(((OptionCacheControl) option).cacheControl());
+          setCacheControl = true;
+        } else if (option instanceof OptionBlockSize) {
+          blockSize = ((OptionBlockSize) option).size();
+        } else {
+          throw new UnsupportedOperationException(option.toString());
         }
-        // The rest will be interpreted later.
       } else {
         throw new UnsupportedOperationException(option.toString());
       }
     }
+
     CloudStoragePath fromPath = checkPath(source);
-    CloudStoragePath toPath = checkPath(target);
 
     blockSize = blockSize != -1 ? blockSize
         : Ints.max(fromPath.getFileSystem().config().blockSize(),
```

Comment:
yes, after we get the basics working I'd like to implement blockSize and benchmark its impact.